judge/models/problem.py

```diff
@@ -535,3 +558,38 @@ class Meta:
         )
         verbose_name = _('solution')
         verbose_name_plural = _('solutions')
+
+
+class ProblemPointsVote(models.Model):
+    points = models.IntegerField(
+        verbose_name=_('proposed points'),
+        help_text=_('the amount of points you think this problem deserves.'),
+        validators=[
+            MinValueValidator(settings.DMOJ_PROBLEM_MIN_USER_POINTS_VOTE),
+            MaxValueValidator(settings.DMOJ_PROBLEM_MAX_USER_POINTS_VOTE),
+        ],
+    )
+
+    voter = models.ForeignKey(Profile, related_name='problem_points_votes', on_delete=CASCADE)
+    problem = models.ForeignKey(Problem, related_name='problem_points_votes', on_delete=CASCADE)
+    vote_time = models.DateTimeField(
+        verbose_name=_('vote time'),
+        help_text=_('the time this vote was cast'),
+        auto_now_add=True,
+        blank=True,
+    )
+
+    note = models.TextField(
+        verbose_name=_('note'),
+        help_text=_('justification for problem point value.'),
+        max_length=2048,
+        blank=True,
+        default='',
+    )
+
+    class Meta:
+        verbose_name = _('vote')
+        verbose_name_plural = _('votes')
+
+    def __str__(self):
+        return f'{self.voter}: {self.points} for {self.problem.code}'
```

Comment:
```suggestion
        return f'Points vote by {self.voter} for {self.problem}'
```
I would say to use this instead so that the string for an object does not change when the user updates their proposed point value. Also makes it more inline with the rest of DMOJ's `__str__`.